### PR TITLE
feat(swc/plugin): support trace for plugin execution

### PR DESCRIPTION
**Pull request by kwonoj**
**Description:**

Finally, I can connect the pieces together. 🎉 

One of the major reason I tried to push forward runtime tracing support is to diagnost user-level execution. Plugin is an exact fit, I wanted to have some answers like `swc is being slow cause plugin xxxx taking too long`. I added initial trace for those in this PR.

image

This tells us

- What plugin module takes X time for total execution
- Serialization cost (loading, json serialization for config, context..)
- Actual plugin transformation cost

and we can add more gradually.

**Related issue (if exists):**

### CODE REVIEW

File: crates/swc/src/plugin.rs [hunk 1]
```diff
@@ -86,6 +87,13 @@ impl RustPlugins {
         // transform.
         if let Some(plugins) = &self.plugins {
             for p in plugins {
+                let span = tracing::span!(
```
Comment (kwonoj) [inline crates/swc/src/plugin.rs:90]: Crateing manual span to track time to `serialize` json vs. actual plugin transform execution.

**Comment by kwonoj** [comment]
This still doesn't able to explain plugin's internal stage. Since it's running under wasm runtime, I doubt we can have easy way to have integrated traces even.

File: crates/swc_ecma_transforms_compat/Cargo.toml [hunk 2]
```diff
@@ -34,7 +34,7 @@ swc_ecma_transforms_classes = {version = "0.50.0", path = "../swc_ecma_transform
 swc_ecma_transforms_macros = {version = "0.3.0", path = "../swc_ecma_transforms_macros"}
 swc_ecma_utils = {version = "0.68.0", path = "../swc_ecma_utils"}
 swc_ecma_visit = {version = "0.54.0", path = "../swc_ecma_visit"}
-swc_trace_macro = {versio = "0.1.0", path = "../swc_trace_macro"}
+swc_trace_macro = {version = "0.1.0", path = "../swc_trace_macro"}
```
Comment (kwonoj) [inline crates/swc_ecma_transforms_compat/Cargo.toml:37]: this seems typo, fixed in this PR.

File: .gitignore [hunk 3]
```diff
@@ -52,4 +52,6 @@ pkg/
 *.tmp.js
 
 # Used to see output of babel.
-/lab
\ No newline at end of file
+/lab
+
+*.mm_profdata
```
Comment (kwonoj) [inline .gitignore:57]: unsure if it's intended but `npm run build:dev` always run `-- -Zself-profile`. Either case I don't expect we'll check in this profile ever.

**Review by kdy1 — APPROVED** [review]
LGTM.
Thanks!

---

swc-bump:
 - swc
 - swc_plugin_runner
 - swc_ecma_transforms_compat

**Pull request merged**